src/analytics/ledger/unlock_conditions.rs

```diff
@@ -19,41 +19,45 @@ pub struct UnlockConditionAnalytics {
     pub storage_deposit_return_inner_value: u64,
 }
 
-impl UnlockConditionAnalytics {
-    /// Initialize the analytics by reading the current ledger state.
-    pub fn init<'a>(unspent_outputs: impl IntoIterator<Item = &'a LedgerOutput>) -> Self {
-        let mut measurement = Self::default();
-        for output in unspent_outputs {
-            match &output.output {
-                Output::Alias(_) => {}
-                Output::Basic(basic) => {
-                    if basic.timelock_unlock_condition.is_some() {
-                        measurement.timelock += output;
-                    }
-                    if basic.expiration_unlock_condition.is_some() {
-                        measurement.expiration += output;
-                    }
-                    if let Some(storage) = basic.storage_deposit_return_unlock_condition {
-                        measurement.storage_deposit_return += output;
-                        measurement.storage_deposit_return_inner_value += storage.amount.0;
-                    }
+fn add_unspent_outputs<'a>(measurement: &mut UnlockConditionAnalytics, unspent_outputs: impl IntoIterator<Item = &'a LedgerOutput>) {
```

Comment:
Why isn't this a method on UnlockConditionAnalytics?

Reply:
Fair point

Reply:
I like free functions though